src/processor/map/map_delete.cpp

```diff
@@ -12,107 +14,98 @@ using namespace kuzu::storage;
+    auto info = NodeDeleteInfo(boundInfo.deleteType, nodeIDPos);
+    if (node.isMultiLabeled()) {
+        common::table_id_map_t<ExtraNodeDeleteInfo> extraInfos;
+        for (auto id : node.getTableIDs()) {
+            auto pkPos = getDataPos(*node.getPrimaryKey(id), schema);
+            extraInfos.insert({id, getExtraNodeDeleteInfo(id, pkPos)});
         }
-        return std::make_unique<SingleLabelNodeDeleteExecutor>(table, std::move(fwdRelTables),
-            std::move(bwdRelTables), info->deleteType, nodeIDPos);
+        return std::make_unique<MultiLabelNodeDeleteExecutor>(std::move(info),
+            std::move(extraInfos));
+    }
+    auto pkPos = getDataPos(*node.getPrimaryKey(node.getSingleTableID()), schema);
+    auto extraInfo = getExtraNodeDeleteInfo(node.getSingleTableID(), pkPos);
+    return std::make_unique<SingleLabelNodeDeleteExecutor>(std::move(info), std::move(extraInfo));
+}
+
+std::unique_ptr<PhysicalOperator> PlanMapper::mapDelete(planner::LogicalOperator* logicalOperator) {
+    auto delete_ = logicalOperator->constPtrCast<LogicalDelete>();
+    switch (delete_->getTableType()) {
+    case TableType::NODE: {
+        return mapDeleteNode(logicalOperator);
+    }
+    case TableType::REL: {
+        return mapDeleteRel(logicalOperator);
+    }
+    default:
+        KU_UNREACHABLE;
     }
 }
 
 std::unique_ptr<PhysicalOperator> PlanMapper::mapDeleteNode(LogicalOperator* logicalOperator) {
-    auto logicalDeleteNode = (LogicalDeleteNode*)logicalOperator;
-    auto inSchema = logicalDeleteNode->getChild(0)->getSchema();
+    auto delete_ = logicalOperator->constPtrCast<LogicalDelete>();
+    auto inSchema = delete_->getChild(0)->getSchema();
     auto prevOperator = mapOperator(logicalOperator->getChild(0).get());
     std::vector<std::unique_ptr<NodeDeleteExecutor>> executors;
-    for (auto deleteInfo : logicalDeleteNode->getInfos()) {
-        executors.push_back(getNodeDeleteExecutor(deleteInfo, *inSchema, clientContext));
+    for (auto& info : delete_->getInfos()) {
+        executors.push_back(getNodeDeleteExecutor(info, *inSchema));
     }
     return std::make_unique<DeleteNode>(std::move(executors), std::move(prevOperator),
-        getOperatorID(), logicalDeleteNode->getExpressionsForPrinting());
+        getOperatorID(), delete_->getExpressionsForPrinting());
 }
 
-static std::unique_ptr<RelDeleteExecutor> getRelDeleteExecutor(
-    storage::StorageManager& storageManager, const binder::RelExpression& rel,
-    const Schema& inSchema) {
-    auto srcNodePos = DataPos(inSchema.getExpressionPos(*rel.getSrcNode()->getInternalID()));
-    auto dstNodePos = DataPos(inSchema.getExpressionPos(*rel.getDstNode()->getInternalID()));
-    auto relIDPos = DataPos(inSchema.getExpressionPos(*rel.getInternalIDProperty()));
+std::unique_ptr<RelDeleteExecutor> PlanMapper::getRelDeleteExecutor(const BoundDeleteInfo& info,
+    const Schema& schema) const {
+    auto sm = clientContext->getStorageManager();
```

Comment:
hmm is abbr. necessary for `storageManager`? this doesn't like a common abbr to me. I'd rather stay with `storageManager`.